JavaScript/TypeScript.sublime-syntax

```diff
@@ -397,7 +397,7 @@ contexts:
             scope: variable.other.readwrite.js
             push:
               - match: in{{identifier_break}}
-                scope: keyword.operator.type.js
+                scope: keyword.operator.type.in.js
```

Comment:
You are right with `keyword.operator.comparison` probably being a better choice. 

I can't really follow operator's being rather messy, though. They may probably not yet be perfect, but mainly because we've not yet touched all syntaxes in detail. Type specific operators and keywords haven't been standardized yet and there might probably always be syntax specific special cases not fitting into any existing scheme, but well.

The discussion about `in` was probably made with regards to Python, as this is what it does. It scopes them differently depending on statement. I don't really find that useful and some other syntaxes already follow the `keyword.operator` scheme.

Reply:
This particular `in` isn't the regular JS `in` operator (which is unchanged in TS), but rather special syntax for mapped types:

```ts
type FlagSet<Keys extends string> = {
  [K in Keys]: boolean
}
```

Honestly, it's not really an operator in the conventional sense. But to the extent that we treat it like one, it *is* type-specific, and it's not a comparison.

It should probably have the same scope as the `in` or `of` in a for loop:

```js
for (const x in y) {}
for (const x of y) {}
```

Both of those are currently `keyword.operator.word`, which is also questionable. By contrast, the actual `in` operator is currently `keyword.operator`, with no `word`.

If I picked a brand-new scope today for `in`/`of` in for loops in mapped types, I might pick `keyword.declaration`. I don't have a strong opinion about changing it now or what to change it to, but I acknowledge that the current scopes are a bit silly.

Reply:
I somehow had python's `if foo in [a,b,c]` statement in mind, when it came to `logical` vs. `comparison` scopes.

I wonder if it makes sense to introduce a `keyword.operator.iterator` scope for such keywords in loops. It might even be suitable in test statements such as the one above, if the forementioned scopes are not desired.